src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -17,30 +17,34 @@ import {requestExternalStoragePermission} from '../../../utils/permission';
 import {setChannel} from '../../../context/actions/setChannel';
 import {setParticipants} from '../../../context/actions/groupChat';
 
-const useGroupInfo = () => {
-  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
+const useGroupInfo = (channelId = null) => {
   const navigation = useNavigation();
-  const {participants, asset} = groupChatState;
+
+  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
+  const [, dispatchChannel] = React.useContext(Context).channel;
   const [client] = React.useContext(Context).client;
   const [channelState] = React.useContext(Context).channel;
   const [profile] = React.useContext(Context).profile;
-  const {channel, profileChannel} = channelState;
+
+  const blockModalRef = React.useRef(null);
+
   const [isLoadingMembers, setIsLoadingMembers] = React.useState(false);
   const [uploadedImage, setUploadedImage] = React.useState('');
   const [isUploadingImage, setIsUploadingImage] = React.useState(false);
   const [isLoadingInitChat, setIsLoadingInitChat] = React.useState(false);
   const [username, setUsername] = React.useState(channelState.channel?.data?.name);
-  const createChat = channelState.channel?.data?.created_at;
-  const countUser = Object.entries(participants).length;
   const [selectedUser, setSelectedUser] = React.useState(null);
   const [newParticipant, setNewParticipant] = React.useState([]);
   const [openModal, setOpenModal] = React.useState(false);
   const [isAnonymousModalOpen, setIsAnonymousModalOpen] = React.useState(false);
   const [isFetchingAllowAnonDM, setIsFetchingAllowAnonDM] = React.useState(false);
-  const [, dispatchChannel] = React.useContext(Context).channel;
+
   const {createSignChat, handleAnonymousMessage} = useCreateChat();
-  const blockModalRef = React.useRef(null);
 
+  const {participants, asset} = groupChatState;
+  const countUser = Object.entries(participants).length;
+  const {channel, profileChannel} = channelState;
+  const createChat = channelState.channel?.data?.created_at;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The order of the state variables and hooks has been changed. While this doesn't affect the functionality, it can make the code harder to read and maintain. It's generally a good practice to keep related pieces of state and logic grouped together for better readability and maintainability.

```diff
-  const navigation = useNavigation();
-  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
-  const [, dispatchChannel] = React.useContext(Context).channel;
-  const [client] = React.useContext(Context).client;
-  const [channelState] = React.useContext(Context).channel;
-  const [profile] = React.useContext(Context).profile;
-  const blockModalRef = React.useRef(null);
-  const [isLoadingMembers, setIsLoadingMembers] = React.useState(false);
-  const [uploadedImage, setUploadedImage] = React.useState('');
-  const [isUploadingImage, setIsUploadingImage] = React.useState(false);
-  const [isLoadingInitChat, setIsLoadingInitChat] = React.useState(false);
-  const [username, setUsername] = React.useState(channelState.channel?.data?.name);
-  const [selectedUser, setSelectedUser] = React.useState(null);
-  const [newParticipant, setNewParticipant] = React.useState([]);
-  const [openModal, setOpenModal] = React.useState(false);
-  const [isAnonymousModalOpen, setIsAnonymousModalOpen] = React.useState(false);
-  const [isFetchingAllowAnonDM, setIsFetchingAllowAnonDM] = React.useState(false);
-  const {createSignChat, handleAnonymousMessage} = useCreateChat();
-  const {participants, asset} = groupChatState;
-  const countUser = Object.entries(participants).length;
-  const {channel, profileChannel} = channelState;
-  const createChat = channelState.channel?.data?.created_at;
-  const anonUserEmojiName = channelState?.channel?.data?.anon_user_info_emoji_name;
+  const navigation = useNavigation();
+  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
+  const {participants, asset} = groupChatState;
+  const countUser = Object.entries(participants).length;
+  const [client] = React.useContext(Context).client;
+  const [channelState] = React.useContext(Context).channel;
+  const {channel, profileChannel} = channelState;
+  const createChat = channelState.channel?.data?.created_at;
+  const anonUserEmojiName = channelState?.channel?.data?.anon_user_info_emoji_name;
+  const [profile] = React.useContext(Context).profile;
+  const [, dispatchChannel] = React.useContext(Context).channel;
+  const blockModalRef = React.useRef(null);
+  const [isLoadingMembers, setIsLoadingMembers] = React.useState(false);
+  const [uploadedImage, setUploadedImage] = React.useState('');
+  const [isUploadingImage, setIsUploadingImage] = React.useState(false);
+  const [isLoadingInitChat, setIsLoadingInitChat] = React.useState(false);
+  const [username, setUsername] = React.useState(channelState.channel?.data?.name);
+  const [selectedUser, setSelectedUser] = React.useState(null);
+  const [newParticipant, setNewParticipant] = React.useState([]);
+  const [openModal, setOpenModal] = React.useState(false);
+  const [isAnonymousModalOpen, setIsAnonymousModalOpen] = React.useState(false);
+  const [isFetchingAllowAnonDM, setIsFetchingAllowAnonDM] = React.useState(false);
+  const {createSignChat, handleAnonymousMessage} = useCreateChat();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->